### DISCUSSION TITLE
Small Submod - Spawning a Crashed Chopper at Player's Location

### DISCUSSION THREAD

**Discussion by JustASausage** [TEHE-Studios/ExpandedHelicopterEvents#332] [Q&A]
I am a dev who has never worked in lua or done any modding. That said, I had some fun poking around this mod's code and think I figured out how to do what I want to do which is Spawn a Crashed Chopper at the Player's location. The problem I am facing is that I do not know how to set up the submod to actually work. 

What I did was add a new function to DebugTests in the same vein as the launchHeliTest functions. Mine, however, specifically spawns the military helicopter, sets its position to directly on the Player, then tells it to crash.

My problem is that I do not know how to do this in a way that it actually works (lol)! I understand that when my server launches, it verifies all the mods and when it sees that my mod file has been edited, it fails to launch. How do I set this up as a submod that I can run on my server?

Alternatively, is there a simple way to spawn the crashed helicopter object manually and bypass this issue altogether? My end goal is simply to set up some fun scenarios for me and my friends.

**Comment by Chuckleberry-Finn** [comment]
If you're the admin of the server you can join in with debug mode, and send a military chopper to your location, toggling on close-by, and always crash, should produce a crashed event of your choosing.

**Comment by JustASausage** [comment]
Thanks for the response! I have tried that approach and it is not what I am looking for. If I toggle All Crash before launching the chopper, it crashes immediately, meaning I cannot put it where I want to. If I launch the chopper and have MoveToPlayer toggled, it flies to me, but does not crash when I set All Crash while the chopper exists.

If "close-by" is another option I am missing, please let me know where I can find that as I do not see it in the debug menu.

**Reply by Chuckleberry-Finn** [reply]
I forget the exact name, but it should be near the top, makes it so events spawn closer than the edge of the map.

**Comment by ThoriumLeaf** [comment]
Admins can just spawn it like any other vehicle: you can right click a tile > spawn vehicle > search for UH-60 (or maybe UH60 i cant remember) in the dropdown and it will be in there.
